### PR TITLE
feat: publish test result

### PR DESCRIPTION
**Pull request by TranQuangPhi**
## Check List
- [ ] Your code builds clean without any errors or warnings.
- [ ] Check coding convention.
- [ ] Updated Storybook components and stories to reflect any changes made to the UI.
- [ ] Test covers all edge cases and coverage is greater than or equal to 80%.

## Proof of Completeness
Screenshot changes introduced" width="250" />
OR
Video link

### CODE REVIEW

File: .github/workflows/publish-test-result.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+name: Publish Tests Result
```
Comment (honghoangsts) [inline .github/workflows/publish-test-result.yml:1]: Change name to `Run unit test`

**Review by honghoangsts — CHANGES_REQUESTED** [review]

File: .github/workflows/publish-test-result.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          node-version: '16'
+
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Run unit tests and capture results
+        id: unit-test-results
+        run: |
+          yarn test --json --outputFile=test-results.json
+          cat test-results.json
```
Comment (baonguyenhsts) [inline .github/workflows/publish-test-result.yml:30]: Do we need this line?

File: .github/workflows/publish-test-result.yml (hunk 1 above)
Comment (TranQuangPhi) [reply .github/workflows/publish-test-result.yml]: done

File: .github/workflows/publish-test-result.yml (hunk 2 above)
Comment (TranQuangPhi) [reply .github/workflows/publish-test-result.yml]: Oh yeah, we don't need it. I forgot to remove it when I was using it for testing. Thanks for reminding me

File: .github/workflows/publish-test-result.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+      - name: Publish test results
+        uses: actions/github-script@v6
+        if: ${{ github.event_name == 'pull_request' && github.event.pull_request }}
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const {repo, owner} = context.repo
+            const fs = require('fs');
+            const results = JSON.parse(fs.readFileSync('test-results.json', 'utf8'));
```
Comment (baonguyenhsts) [inline .github/workflows/publish-test-result.yml:39]: Could we get the code coverage of a test file from the result? If yes, please show it in the Test Results like this format.
[Test format](https://user-images.githubusercontent.com/62632321/218383185-0581c50c-2c18-4429-8b93-10d2a2325714.png)

**Comment by ThinhKimVo** [comment]
should change workflow file name follow a format **action_[actionName]**, **ci_[actionName]**. ex: `action_run_unit_test`

**Comment by baonguyenhsts** [comment]
> should change workflow file name with a format **action_[actionName]**, **ci_[actionName]**. ex: `action_run_unit_test`

Hi @ThinhKimVo,
Did you mean to change the name of the yml file from publish-test-result to action_run_unit_test?

**Comment by TKimVo** [comment]
> > should change workflow file name with a format **action_[actionName]**, **ci_[actionName]**. ex: `action_run_unit_test`
> 
> Hi @ThinhKimVo, Did you mean to change the name of the yml file from publish-test-result to action_run_unit_test?

Yes, the name should start with a prefix action_ to clear

**Comment by anhle10051996** [comment]
Hi @ThinhKimVo, the workflow file name is updated

**Review by baonguyenhsts — APPROVED** [review]

**Review by ThinhKimVo — APPROVED** [review]

**Review by loc-nguyenthien — APPROVED** [review]

**Review by huydosgtech — APPROVED** [review]
